gtk4/src/constraint_layout.rs

```diff
@@ -0,0 +1,44 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Constraint, ConstraintLayout, Widget};
+use glib::translate::*;
+use glib::{Cast, IsA, Object};
+use std::collections::HashMap;
+
+impl ConstraintLayout {
+    #[doc(alias = "gtk_constraint_layout_add_constraints_from_descriptionv")]
+    #[doc(alias = "gtk_constraint_layout_add_constraints_from_description")]
+    #[doc(alias = "add_constraints_from_descriptionv")]
+    pub fn add_constraints_from_description<W: IsA<Widget>>(
+        &self,
+        lines: &[&str],
+        hspacing: i32,
+        vspacing: i32,
+        views: HashMap<&str, &W>,
+    ) -> Result<Vec<Constraint>, glib::Error> {
+        unsafe {
+            let err = std::ptr::null_mut();
+            // Because we only have a ToGlibPtr implemented for HashMap<&str, Object>
+            // and we want a simpler API that makes use of a Widget directly
+            let views_objs: HashMap<&str, Object> = HashMap::with_capacity(views.capacity());
+            for (key, widget) in views.into_iter() {
+                views_objs.insert(key, widget.unsafe_cast::<Object>());
+            }
```

Comment:
`let view_objs = views.into_iter().map(|(k, v)| (k, v.unsafe_cast::<Object>())).collect::<HashMap<_, _>>();`

But you probably:
  * Don't want to pass the hashmap by value but instead by reference (doesn't change much)
  * Directly build the `GHashTable` because going via an intermediate hash map is quite some churn

Reply:
Oh I didn't really thought of building the GHashTable directly here... sounds much better than re-building the map two times :)